### PR TITLE
Add recipe for QuantEcon package

### PR DESCRIPTION
**Pull request by mmcky**
This will add the build recipe for the QuantEcon package. 

http://quantecon.org/python_index.html

### CODE REVIEW

**Comment by mmcky** [comment]
Conda Forge Team. 

```python
+ python setup.py install --single-version-externally-managed --record record.txt
usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help

error: option --single-version-externally-managed not recognized
```

I was following instructions in the default template. Should I not use this option? Thanks.

**Comment by jakirkham** [comment]
It looks like your package does not use `setuptools`. Please drop it from `build` and drop those options too.

**Comment by mmcky** [comment]
Thanks @jakirkham.

File: recipes/quantecon/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    # - quantecon.markov
+    # - quantecon.random
+    # - quantecon.util
+
+about:
+  home: https://github.com/QuantEcon/QuantEcon.py
+  # Remember to specify the license variants for BSD, Apache, GPL, and LGLP.
+  # Prefer the short version, e.g: GPL-2.0 instead of GNU General Public License version 2.0
+  # See https://opensource.org/licenses/alphabetical
+  license: BSD
```
Comment (jakirkham) [inline recipes/quantecon/meta.yaml]: Please specify license version (e.g. `BSD 3-Clause`).

**Comment by jakirkham** [comment]
Of course.

Still some things to do here. Mainly it would be good to remove the extraneous comments. We provide them in the example to help, but they have served their value at this point. So we can drop them.

**Comment by mmcky** [comment]
Thanks for the comments -- tidying these items up now. 

Can the ``description`` field in the ``about`` section contain markdown?

File: recipes/quantecon/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    1. `QuantEcon Course Website <http://quant-econ.net>`_
+
+  doc_url: http://quanteconpy.readthedocs.io/en/latest/
+  dev_url: https://github.com/QuantEcon/QuantEcon.py
+
+extra:
+  recipe-maintainers:
+    - mmcky
+
+# See
```
Comment (patricksnape) [inline recipes/quantecon/meta.yaml]: You can remove this.

**Comment by mmcky** [comment]
Thanks @patricksnape I have reviewed the description and simplified it further. Once we get this package into the ``conda-forge`` channel i will update install instructions away from ``pip`` in favour of ``conda``.

**Comment by mmcky** [comment]
Not sure why ``AppVeyor`` is not longer a test.

File: recipes/quantecon/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  number: 0
+  script: python setup.py install
+
+requirements:
+  build:
+    - python
+
+  run:
+    - python
+    - sphinx
```
Comment (patricksnape) [inline recipes/quantecon/meta.yaml]: Do you build the docs are part of your build? Or use sphinx at runtime?

**Comment by mmcky** [comment]
Thanks -- good pickup. I have reviewed the run time dependencies list.

File: recipes/quantecon/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,71 @@
+{% set version = "0.3.3" %}
+{% set md5 = "11df25c2f489a3675bb574076990ad6e" %}
+{% set sha256 = "0aa5f4d3d4b6a92982fb4172f5c1d8c4e1af7abf185b23f103a3ed95520c5730" %}
+
+package:
+  name: quantecon
+  version: {{ version }}
+
+source:
+  # GitHub Downloads
```
Comment (jakirkham) [inline recipes/quantecon/meta.yaml]: This seems self-evident, right?

**Comment by jakirkham** [comment]
Thanks @mmcky. Looks like there are a couple easy things left to do and then this is good to go.

File: recipes/quantecon/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  doc_url: http://quanteconpy.readthedocs.io/en/latest/
+  dev_url: https://github.com/QuantEcon/QuantEcon.py
+
+extra:
+  recipe-maintainers:
+    - mmcky
+
+# See
+# http://docs.continuum.io/conda/build.html for
+# more information about meta.yaml
```
Comment (ocefpaf) [inline recipes/quantecon/meta.yaml]: @mmcky just remove the remaining boilerplate comments from `conda-skeleton` and this is good to go!

**Comment by mmcky** [comment]
How do I restrict the compilation to ``python=3.5``? This currently isn't a problem but we are starting to use the ``@`` symbol for matrix multiplication and we are planning to only support ``python=3.5+``.? Thanks.

File: recipes/quantecon/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  version: {{ version }}
+
+source:
+  fn: QuantEcon.py-{{ version }}.tar.gz
+  url: https://github.com/QuantEcon/QuantEcon.py/archive/{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: python setup.py install
```
Comment (ocefpaf) [inline recipes/quantecon/meta.yaml]: > How do I restrict the compilation to python=3.5 for CI?

You can add

```yaml
skip: True  # [not py35]
```

here in the `build` field.

**Comment by mmcky** [comment]
ahh -- thanks @ocefpaf. I will do that now.

**Comment by jakirkham** [comment]
I think it would need to be `# [not (py27 or py34)]`.

**Comment by jakirkham** [comment]
However, for what you want to achieve there wouldn't even be a `not`. So the full thing should be.

`skip: True  # [py27 or py34]`

Better still might be something like

`skip: True  # [py<35]`

However, I'd have to double check that the latter doesn't run into issue ( https://github.com/conda-forge/conda-smithy/issues/257 ).

**Comment by mmcky** [comment]
ahh I see. 
I tried checking out the conda documentation and couldn't find any info on this ``selector`` syntax. 
Thanks for the help -- really appreciate it.

**Comment by jakirkham** [comment]
Of course. Yeah, so I tested with `conda-smithy` and the last one ( `skip: True # [py<35]` ) works. So that would be my suggestion if you are concerned about future Python versions.

File: recipes/quantecon/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    Assuming you have `pip <https://pypi.python.org/pypi/pip>`_ on your computer --- as will be the case if you've `installed Anaconda <http://quant-econ.net/getting_started.html#installing-anaconda>`_ --- you can install the latest stable release of ``quantecon`` by typing
+        pip install quantecon
+    at a terminal prompt
+    **Note:** There is also a Julia version available for Julia users `QuantEcon.jl <https://github.com/QuantEcon/QuantEcon.jl>`_
+  doc_url: http://quanteconpy.readthedocs.io/en/latest/
+  dev_url: https://github.com/QuantEcon/QuantEcon.py
+
+extra:
+  recipe-maintainers:
+    - mmcky
```
Comment (jakirkham) [inline recipes/quantecon/meta.yaml]: nit: Please add a terminal newline.

Comment (jakirkham) [inline recipes/quantecon/meta.yaml]: Done.

File: recipes/quantecon/meta.yaml (hunk 6 above)
Comment (jakirkham) [inline recipes/quantecon/meta.yaml]: Done.

**Comment by mmcky** [comment]
Thanks for all of your comments and assistance.

**Comment by jakirkham** [comment]
Thanks for your contribution, @mmcky.

**Comment by jakirkham** [comment]
You should get an email that invite you to join conda-forge, @mmcky. Once accepted you will be added to a team with the same name as this recipe. Those will give you permissions on the feedstock (repo) for this recipe.

Make sure when proposing any change that you go through the typical GitHub workflow of forking the feedstock and making changes in your fork that you PR back. Once merged CIs will build and deploy any changes you make. Please let us know if you have any questions and welcome to conda-forge.

**Pull request merged**